tests/test_examples.py

```diff
@@ -0,0 +1,386 @@
+    for installer, install_dir in create_installer(input_path, tmp_path):
+        _run_installer(input_path, installer, install_dir, request=request)
+
+
+def test_example_customized_welcome_conclusion(tmp_path, request):
+    input_path = _example_path("customized_welcome_conclusion")
+    for installer, install_dir in create_installer(input_path, tmp_path):
+        _run_installer(input_path, installer, install_dir, request=request)
+
+
+def test_example_extra_envs(tmp_path, request):
+    input_path = _example_path("extra_envs")
+    for installer, install_dir in create_installer(input_path, tmp_path):
+        _run_installer(input_path, installer, install_dir, request=request)
+
+
+def test_example_extra_files(tmp_path, request):
+    input_path = _example_path("extra_files")
+    for installer, install_dir in create_installer(input_path, tmp_path, with_spaces=True):
+        _run_installer(input_path, installer, install_dir, request=request)
+
+
+def test_example_miniforge(tmp_path, request):
+    input_path = _example_path("miniforge")
+    for installer, install_dir in create_installer(input_path, tmp_path):
+        if installer.suffix == ".sh":
+            # try both batch and interactive installations
+            install_dirs = install_dir / "batch", install_dir / "interactive"
+            installer_inputs = None, f"\nyes\n{install_dir / 'interactive'}\nno\n"
+        else:
+            install_dirs = (install_dir,)
+            installer_inputs = (None,)
+        for installer_input, install_dir in zip(installer_inputs, install_dirs):
+            _run_installer(
+                input_path,
+                installer,
+                install_dir,
+                installer_input=installer_input,
+                # PKG installers use their own install path, so we can't check sentinels
+                # via `install_dir`
+                check_sentinels=installer.suffix != ".pkg",
+            )
+            if installer.suffix == ".pkg" and ON_CI:
+                _sentinel_file_checks(input_path, Path(os.environ["HOME"]) / "Miniforge3")
+
+
+def test_example_noconda(tmp_path, request):
+    input_path = _example_path("noconda")
+    for installer, install_dir in create_installer(input_path, tmp_path, with_spaces=True):
+        _run_installer(input_path, installer, install_dir, request=request)
+
+
+@pytest.mark.skipif(sys.platform != "Darwin", reason="macOS only")
+def test_example_osxpkg(tmp_path, request):
+    input_path = _example_path("osxpkg")
+    for installer, install_dir in create_installer(input_path, tmp_path):
+        _run_installer(input_path, installer, install_dir, request=request)
+
+
+def test_example_scripts(tmp_path, request):
+    input_path = _example_path("scripts")
+    for installer, install_dir in create_installer(input_path, tmp_path, with_spaces=True):
+        _run_installer(input_path, installer, install_dir, request=request)
+
+
+def test_example_shortcuts(tmp_path, request):
```

Comment:
For these I would have expected a `parametrize` over the across the `_example_path`s. Assuming that function returns filenames quickly it seems fine to refactor this way and would give more complete failure reporting for example if a change breaks multiple examples. It also would allow selecting tests by the example filename because that would show up in the resulting test name